source/features/prevent-pr-commit-link-loss.tsx

```diff
@@ -1,16 +1,33 @@
-import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
+import AlertIcon from 'octicon/alert.svg';
+import delegate from 'delegate-it';
+import React from 'dom-chef';
+import select from 'select-dom';
 
 import features from '../libs/features';
-import {preventPrCommitLinkBreak} from '../libs/utils';
+import {containsPrCommitLink, preventPrCommitLinkBreak} from '../libs/utils';
 
 function init(): void {
-	delegate(document, 'form#new_issue, form.js-new-comment-form', 'submit', updateTextArea);
+	delegate(document, 'form#new_issue textarea, form.js-new-comment-form textarea', 'input', handleTextAreaChange);
 }
 
-function updateTextArea(event: delegate.Event): void {
-	const field = event.delegateTarget.querySelector('textarea')!;
-	field.value = preventPrCommitLinkBreak(field.value);
+function handleTextAreaChange(event: delegate.Event): void {
+	const field = (event.delegateTarget as HTMLInputElement);
+	const fieldValue = field.value;
+
+	if(containsPrCommitLink(fieldValue) && !select.exists('#fix-pr-commit-links')) {
+		const closestForm = field.closest('form')!;
+		select('.form-actions', closestForm)?.prepend(
```

Comment:
Since we expect `.form-actions` to always exist, if it doesn't it's a problem. `?.` silences this kind of problem.

Also form fiels have a `form` property
```suggestion
		select('.form-actions', field.form!)!.prepend(
```